ph5/utilities/tests/test_metadatatoph5.py

```diff
@@ -1,47 +1,46 @@
 '''
 Tests for metadatatoph5
 '''
-
+import os
+import sys
 import unittest
-from ph5.utilities import metadatatoph5
-from ph5.utilities import initialize_ph5
+
 from obspy.core import inventory
 from obspy import UTCDateTime
-import os
-import sys
 from mock import patch
-from ph5.core.tests.test_base import initialize_ex
+from testfixtures import OutputCapture
 
+from ph5.utilities import metadatatoph5
+from ph5.core.tests.test_base import LogTestCase, TempDirTestCase, \
+     initialize_ex
 
-class TestMetadatatoPH5(unittest.TestCase):
 
+class TestMetadatatoPH5(TempDirTestCase, LogTestCase):
     def setUp(self):
-        self.path = 'ph5/test_data/miniseedph5'
-        try:
-            os.mkdir(self.path)
-        except OSError:
-            print ("Creation of the directory %s failed" % self.path)
-        else:
-            print ("Successfully created the directory %s " % self.path)
+        super(TestMetadatatoPH5, self).setUp()
+        if self._testMethodName != 'test_main':
+            self.ph5_object = initialize_ex('master.ph5', self.tmpdir, True)
+            self.metadata = metadatatoph5.MetadatatoPH5(
+                self.ph5_object)
 
-        self.ph5_object = initialize_ex('master.ph5', self.path, True)
-        default_receiver_t = initialize_ph5.create_default_receiver_t()
-        initialize_ph5.set_receiver_t(default_receiver_t)
-        os.remove(default_receiver_t)
-        self.metadata = metadatatoph5.MetadatatoPH5(
-            self.ph5_object)
+    def tearDown(self):
+        if self._testMethodName != 'test_main':
+            self.ph5_object.ph5close()
```

Comment:
The condision excludes test_main(). 

Anyway I think should put 'self.' in front of 'ph5_object' in test_main() so that I can clean it up in tearDown() and throw that condition away.